EventFilter/Utilities/src/json_reader.cpp

```diff
@@ -430,7 +430,7 @@ namespace Json {
       while (token.type_ == tokenComment && ok) {
         ok = readToken(token);
       }
-      bool badTokenType = (token.type_ == tokenArraySeparator || token.type_ == tokenArrayEnd);
+      bool badTokenType = (token.type_ == tokenArraySeparator && token.type_ == tokenArrayEnd);
       if (!ok || badTokenType) {
```

Comment:
This does not make sense, how can this condition be true. As it will always be false so better to remove it

Reply:
You're right. I bisected and found what broke the code, but didn't spot that original condition is always false.

Reply:
jsoncpp seems to support comment at the end of array, which is non-standard. The check is part of this (and buggy). That feature is not relevant for what it is used in Event Filter code, so I dropped the condition with another commit.